torch/csrc/jit/passes/symbolic_shape_analysis.cpp

```diff
@@ -529,13 +533,235 @@ struct SymbolicShapeNodeAnalyzer {
   Node* node_;
 };
 
-void PropagateShapesWithShapeFunction(
+std::shared_ptr<Graph> PropagateShapesWithShapeFunction(
     Node* n,
     std::shared_ptr<Graph>& shape_compute_graph,
     const AliasDb& db) {
-  SymbolicShapeNodeAnalyzer(n, shape_compute_graph, db).run();
+  return SymbolicShapeNodeAnalyzer(n, shape_compute_graph, db).run();
 }
 
+struct SymbolicShapeGraphAnalyzer {
+  SymbolicShapeGraphAnalyzer(
+      std::shared_ptr<Graph>& graph,
```

Comment:
Does this need to be a `shared_ptr`?  It seems like a raw pointer would do fine here since no ownership is being passed.